src/components/dialogs/filter/explicit-naming-filter-form.tsx

```diff
@@ -0,0 +1,162 @@
+import React, { useCallback, useMemo } from 'react';
+import {
+    EQUIPMENT_ID,
+    EQUIPMENT_TYPE,
+    FILTER_TYPE,
+} from '../../utils/field-constants';
+import yup from '../../utils/yup-config';
+import CustomAgGridTable, {
+    ROW_DRAGGING_SELECTION_COLUMN_DEF,
+} from '../../utils/rhf-inputs/ag-grid-table-rhf/custom-ag-grid-table';
+import { useIntl } from 'react-intl';
+import { useWatch } from 'react-hook-form';
+import { FILTER_EQUIPMENTS } from '../commons/criteria-based/criteria-based-utils';
+import Grid from '@mui/material/Grid';
+import SelectInput from '../../utils/rhf-inputs/select-inputs/select-input';
+import { ValueParserParams } from 'ag-grid-community';
+import { toIntOrNullValue } from '../../utils/dialog-utils';
+import { Generator, Load } from '../../../utils/equipment-types';
+import { FilterType } from '../../../utils/elementType';
+
+export const FILTER_EQUIPMENTS_ATTRIBUTES = 'filterEquipmentsAttributes';
+export const DISTRIBUTION_KEY = 'distributionKey';
+
+export const explicitNamingFilterSchema = {
+    [FILTER_EQUIPMENTS_ATTRIBUTES]: yup
+        .array()
+        .of(
+            yup.object().shape({
+                [EQUIPMENT_ID]: yup.string().nullable(),
+                [DISTRIBUTION_KEY]: yup.number().nullable(),
+            })
+        )
+        // we remove empty lines
+        .compact((row) => !row[DISTRIBUTION_KEY] && !row[EQUIPMENT_ID])
+        .when([FILTER_TYPE], {
+            is: FilterType.EXPLICIT_NAMING.id,
+            then: (schema) =>
+                schema
+                    .min(1, 'emptyFilterError')
+                    .test(
+                        'noKeyWithoutId',
+                        'distributionKeyWithMissingIdError',
+                        (array) => {
+                            return !array!.some((row) => !row[EQUIPMENT_ID]);
+                        }
+                    )
+                    .test(
+                        'ifOneKeyThenKeyEverywhere',
+                        'missingDistributionKeyError',
+                        (array) => {
+                            return !(
+                                array!.some((row) => row[DISTRIBUTION_KEY]) &&
+                                array!.some((row) => !row[DISTRIBUTION_KEY])
+                            );
+                        }
+                    ),
+        }),
+};
+
+export const explicitNamingFilterEmptyFormData = {
+    [FILTER_EQUIPMENTS_ATTRIBUTES]: [],
+};
+
+const defaultRowData = {
+    [EQUIPMENT_ID]: null,
```

Comment:
```suggestion
    [EQUIPMENT_ID]: "",
```
empty value for string inputs should be empty string

Reply:
Ok, any reason ?

Reply:
Ok done